test/PHPUnit/Controller/AbstractControllerTestCaseTest.php

```diff
@@ -538,6 +539,66 @@ public function testAssertNotTemplateName(): void
         $this->assertNotTemplateName('template/does/not/exist');
     }
 
+    /**
+     * Test case for a controller returning a view with deeply nested children
+     * View hierarchy:
+     *   layout/layout -> baz/index/childview -> child1 -> child3
+     *                                        -> child2
+     */
+    public function testSearchTemplatesVerifiesDeeplyNestedTemplateName(): void
+    {
+        $this->dispatch('/childview');
+
+        // Check that the rendered content
+        $this->assertQueryContentContains('p', 'Parent');
+        $this->assertQueryContentContains('p', 'Child 1');
+        $this->assertQueryContentContains('p', 'Child 2');
+        $this->assertQueryContentContains('p', 'Child 3');
+
+        $this->assertTemplateName('layout/layout');
+        $this->assertTemplateName('baz/index/childview');
+        $this->assertTemplateName('child1');
+        $this->assertTemplateName('child2');
+        $this->assertTemplateName('child3');
+        $this->assertNotTemplateName('foo');
+    }
+
+    /**
+     * Check that the assertion fails when template is NOT found where it was supposed to found
+     */
+    public function testAssertTemplateNameFailsWhenNotFound(): void
+    {
+        $this->dispatch('/childview');
+
+        try {
+            $this->assertTemplateName('foo');
+            $receivedException = false;
+        } catch (Exception $exception) {
+            $receivedException = true;
```

Comment:
To ensure correct exception was received assert the message content.

This test would fail and uncover the fact that no message is provided in the assertion implementation.